homeassistant/components/climate/venstar.py

```diff
@@ -0,0 +1,287 @@
+https://home-assistant.io/components/climate.venstar/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    STATE_COOL, STATE_HEAT, STATE_IDLE, STATE_AUTO,
+    ClimateDevice, PLATFORM_SCHEMA,
+    SUPPORT_TARGET_TEMPERATURE, SUPPORT_TARGET_HUMIDITY,
+    SUPPORT_FAN_MODE, SUPPORT_OPERATION_MODE,
+    SUPPORT_TARGET_TEMPERATURE_HIGH,
+    SUPPORT_TARGET_TEMPERATURE_LOW,
+    ATTR_TARGET_TEMP_HIGH, ATTR_TARGET_TEMP_LOW)
+
+from homeassistant.const import (
+    CONF_HOST, CONF_PASSWORD, CONF_SSL, CONF_USERNAME, CONF_TIMEOUT,
+    TEMP_FAHRENHEIT, ATTR_TEMPERATURE, TEMP_CELSIUS,
+    PRECISION_WHOLE, STATE_ON)
+
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['venstarcolortouch==0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_SSL = False
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_USERNAME, default=None): cv.string,
+    vol.Optional(CONF_PASSWORD, default=None): cv.string,
+    vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
+    vol.Optional(CONF_TIMEOUT, default=5):
+        vol.All(vol.Coerce(int), vol.Range(min=1))
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Venstar thermostat."""
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    host = config.get(CONF_HOST)
+    if config.get(CONF_SSL):
+        proto = 'https'
+    else:
+        proto = 'http'
+    timeout = config.get(CONF_TIMEOUT)
+
+    import venstarcolortouch
+    client = venstarcolortouch.VenstarColorTouch(addr=host,
+                                                 timeout=timeout,
+                                                 user=username,
+                                                 password=password,
+                                                 proto=proto)
+
+    add_devices([VenstarThermostat(client)])
+
+
+class VenstarThermostat(ClimateDevice):
+    """Representation of a Venstar thermostat."""
+
+    def __init__(self, client):
+        """Initialize the thermostat."""
+        self._client = client
+        self._client.update_info()
+        self._client.update_sensors()
+        self._fan_list = [STATE_ON, STATE_AUTO]
+        self._operation_list = [STATE_HEAT, STATE_COOL, STATE_IDLE, STATE_AUTO]
+
+    def update(self):
+        """Update the data from the thermostat."""
+        _LOGGER.info("Refreshing data from your Venstar Thermostat.")
+        info_success = self._client.update_info()
+        sensor_success = self._client.update_sensors()
+        if not info_success or not sensor_success:
+            _LOGGER.error("Failed to update data from your Thermostat.")
+
+# Thermostat config
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        if self._client.mode == self._client.MODE_AUTO:
```

Comment:
What does this mean? When the device is in `auto` mode, it tries to stay between a min and max temperature? And otherwise… it doesn't?

Reply:
That's correct, yes - the thermostat physically has 4 modes:
- heat- you set a single temperature, and the thermostat will attempt to heat the house UP TO this temp.

-cool- you set a single temperature, and the thermostat will attempt to cool the house DOWN TO this temp

-auto- you set 2 temperatures, and the thermostat will automatically swap between heating and cooling depending upon the temperature of the house.  It's essentially the heat/cool mode combined.

-off/idle - no hold at all, but reporting on temp

I don't expect to use the auto setting, but felt i needed to do something for it in case someone else did.